dotnet/samples/Concepts/Agents/ChatCompletion_HistoryReducer.cs

```diff
@@ -149,25 +166,30 @@ private async Task InvokeChatAsync(ChatCompletionAgent agent, int messageCount)
         }
     }
 
-    private ChatCompletionAgent CreateSummarizingAgent(int reducerMessageCount, int reducerThresholdCount)
+    private ChatCompletionAgent CreateSummarizingAgent(int reducerMessageCount, int reducerThresholdCount, bool useChatClient, out IChatClient? chatClient)
     {
-        Kernel kernel = this.CreateKernelWithChatCompletion();
+        Kernel kernel = this.CreateKernelWithChatCompletion(useChatClient, out chatClient);
+
+        var service = useChatClient
+            ? kernel.GetRequiredService<IChatClient>().AsChatCompletionService()
```

Comment:
The IChatClient still needs to be wrapped as an IChatCompletionService?

Reply:
Yes, for the `ChatCompletionAgent` it does. I cannot change the `ChatCompletionAgent` signature, avoiding introducing `ChatClientAgent` at this stage as I'm avoiding adding new public API's on this migration.